micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jvm/JvmGcMetrics.java

```diff
@@ -204,7 +204,7 @@ private static boolean isManagementExtensionsPresent() {
 
         try {
             Class.forName("com.sun.management.GarbageCollectionNotificationInfo", false,
-                    JvmGcMetrics.class.getClassLoader());
+                    ManagementFactory.getMemoryMXBean().getClass().getClassLoader());
```

Comment:
While this looks like it should work, the `MemoryMXBean` is not related to what this class is doing. Would something like `MemoryPoolMXBean.class.getClassLoader()` or even `ManagementFactory.class.getClassLoader()` work? Are those also loaded by the application container classloader?

Reply:
I tested this with Wildfly and it looks like `MemoryPoolMXBean.class.getClassLoader()` resolves the classloader issue. Let me know if that's not the case for your application server.

Reply:
I just checked it too and it works.

I will update the PR. You want it from 1.1.x branch to 1.1.x branch ?